modules/manage/pages/iceberg/topic-iceberg-integration.adoc

```diff
@@ -435,6 +415,44 @@ FROM (
 
 You can now insert the transformed record back into the main Iceberg table. Redpanda recommends employing a strategy for exactly-once processing to avoid duplicates when reprocessing records.
 
+== Performance considerations
+
+When you enable Iceberg for any substantial workload, you may see most of your cluster's CPU utilization increase as a result of translation of topic data to the Iceberg format. If this process overloads the brokers, Redpanda automatically applies backpressure to clients to prevent runaway lag of the Iceberg table behind the topic log.
+
+=== Use custom partitioning
+
+To improve query performance, consider implementing custom https://iceberg.apache.org/docs/nightly/partitioning/[partitioning^] for the Iceberg topic. Use the xref:reference:properties/topic-properties.adoc#redpanda-iceberg-partition-spec[`redpanda.iceberg.partition.spec`] topic property to define the partitioning scheme:
+
+[,bash,]
+----
+# Create new topic with five topic partitions, replication factor 1, and custom table partitioning for Iceberg
+rpk topic create <new-topic-name> -p5 -r1 -c "redpanda.iceberg.partition.spec=(<partition-key1>, <partition-key2>, ...)"
+----
+
+Valid `<partition-key>` values include a source column name or a transformation of a column. The columns referenced can be Redpanda-defined (such as `redpanda.timestamp`) or user-defined based on a schema that you register for the topic. The Iceberg table stores records that share the same partition key values in the same files based on this specification. 
```

Comment:
> The Iceberg table stores records that share the same partition key values in the same files

Rather, the table stores records with different partition key values in different files (the way it is described now may suggest that it is _guaranteed_ that all records with the same partition key will end up in the single file, which is not true).

Reply:
@ztlpn Would something like `stores records with different partition key values in _separate_ files` convey it well?

Reply:
yes, sounds good!